packages/x-data-grid/src/utils/keyboardUtils.ts

```diff
@@ -57,11 +57,23 @@ export const isHideMenuKey = (key: React.KeyboardEvent['key']) => isTabKey(key)
 export function isPasteShortcut(event: React.KeyboardEvent) {
   if (
     (event.ctrlKey || event.metaKey) &&
-    event.key.toLowerCase() === 'v' &&
+    event.code === 'KeyV' &&
```

Comment:
This can't work, e.g.  event.code assumes a QWERTY keyboard layout which would be wrong with a Dvorak keyboard.